File: 백준/Silver/10773. 제로/제로.py
K = int(input()) #받을 수 있는 총 숫자의 개수 

num = [ ] 

for _ in range(K):
    n = int(input())
    if n == 0:
        num.pop()
    else:
        num.append(n)
    
total_num = sum(num)
print(total_num)

File: 백준/Silver/10773. 제로/제로.js
let input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

let stack = [];
let K = parseInt(input[0]);

for (let i = 1; i <= K; i++) {
  let num = parseInt(input[i]);

  if (num === 0) {
    stack.pop();  
  } else {
    stack.push(num);  
  }
}

let answer = stack.reduce((acc, cur) => acc + cur, 0);

console.log(answer);
